File: 0202-happy-number/0202-happy-number.cpp
class Solution {
public:
    bool isHappy(int n) {
     unordered_set <int> set;
        
        while(n!=1)
        {
            auto pos=set.find(n);
            if(pos!=set.end())
            {
                return false;
            }
            set.insert(n);
            int sum=0;
            while(n>0)
            {
                
                int rem=n%10;
                n/=10;
                sum+=rem*rem;
            }
            n=sum;
        }
        return n==1;
    }
};

File: 0202-happy-number/0202-happy-number.java
class Solution {
    public boolean isHappy(int n) {
        Set<Integer> hs= new HashSet<>();
        
        int sqSum;
        int remain=0;
        
        while(hs.add(n))
        {
            sqSum=0;
            while(n>0)
            {
                remain=n%10;
                sqSum+=remain*remain;
                n/=10;
            }
            if(sqSum==1)
            {
                return true;
            }
            else
            {
                n=sqSum;
            }
        }
        return false;
    }
}